File: server/dotnet/sdk-dotnet-example/Infrastructure/Mappers/PaymentDetails/SepaDirectDebitPaymentMethodSpecificOutputMapper.cs
using SepaDirectDebitPaymentMethodSpecificOutputSdk = OnlinePayments.Sdk.Domain.SepaDirectDebitPaymentMethodSpecificOutput;
using SepaDirectDebitPaymentMethodSpecificOutputDto = Business.Domain.Payments.PaymentDetails.SepaDirectDebitPaymentMethodSpecificOutput;


namespace Infrastructure.Mappers.PaymentDetails;

public static class SepaDirectDebitPaymentMethodSpecificOutputMapper
{
    public static SepaDirectDebitPaymentMethodSpecificOutputDto Map( SepaDirectDebitPaymentMethodSpecificOutputSdk? response)
    {
        return new()
        {
            PaymentProductId = response?.PaymentProductId,
            FraudResults = FraudResultsMapper.Map(response?.FraudResults),
            PaymentProduct771SpecificOutput = PaymentProduct771SpecificOutputMapper.Map(response?.PaymentProduct771SpecificOutput)
        };
    }
}

File: server/nodejs/src/infrastructure/mappers/payment-details/sepa-direct-debit-payment-method-specific-output-mapper.ts
import type { Domain } from 'onlinepayments-sdk-nodejs';
import { SepaDirectDebitPaymentMethodSpecificOutput } from '../../../business/domain/payments/payment-details/sepa-direct-debit-payment-method-specific-output';
import { FraudResultsMapper } from './fraud-results-mapper';
import { PaymentProduct771SpecificOutputMapper } from './payment-product-771-specific-output-mapper';

type SepaDirectDebitPaymentMethodSpecificOutputSdk = Domain.SepaDirectDebitPaymentMethodSpecificOutput;

export const SepaDirectDebitPaymentMethodSpecificOutputMapper = {
    fromSdkResponse: (
        response?: SepaDirectDebitPaymentMethodSpecificOutputSdk | null,
    ): SepaDirectDebitPaymentMethodSpecificOutput | undefined => {
        if (!response) {
            return undefined;
        }

        return {
            paymentProductId: response?.paymentProductId,
            fraudResults: FraudResultsMapper.fromSdkResponse(response?.fraudResults),
            paymentProduct771SpecificOutput: PaymentProduct771SpecificOutputMapper.fromSdkResponse(
                response?.paymentProduct771SpecificOutput,
            ),
        };
    },
};
